### PR TITLE
unistd: add a preliminary implementation of getopt()

### PR DESCRIPTION
**Pull request by Arcterus**
This seems to work, but I haven't tested it thoroughly.  It would probably be a good idea to run it through some more complex situations.  Do we want to implement the GNU extension to allow optional arguments?

### CODE REVIEW

File: src/unistd/src/getopt.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+static mut current_opt: *mut c_char = ptr::null_mut();
+
+#[no_mangle]
+pub unsafe extern "C" fn getopt(
+    argc: c_int,
+    argv: *const *mut c_char,
+    optstring: *const c_char,
+) -> c_int {
+    if current_opt == ptr::null_mut() || *current_opt == 0 {
+        let current_arg = *argv.offset(optind as isize);
```
Comment (dlrobertson) [inline src/unistd/src/getopt.rs]: if `argc` is 1 this could be out of bounds right?

File: src/unistd/src/getopt.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        // XXX: is argc check needed?
+        if optind > argc || current_arg == ptr::null_mut() || *current_arg != b'-' as c_char
```
Comment (dlrobertson) [inline src/unistd/src/getopt.rs]: Wouldn't this be `optind >= argc`?

File: src/unistd/src/getopt.rs (hunk 1 above)
Comment (Arcterus) [reply src/unistd/src/getopt.rs]: Good catch.  I'll fix this later today.

File: src/unistd/src/getopt.rs (hunk 2 above)
Comment (Arcterus) [reply src/unistd/src/getopt.rs]: Yes.

**Comment by Arcterus** [comment]
I didn't even notice I did that lol

**Pull request merged**